### DISCUSSION TITLE
OSC Customization - Background & Icons

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#14491] [Q&A]
Closed moved to the following post 14871

**Comment by guidocella** [comment]
It sounds like you want only the progress bar, you may want to look into https://codeberg.org/NRK/mpv-toolbox/src/branch/master/mfpbar instead of making your own fork.

**Comment by advice2020** [comment]
@guidocella 
Thank you for your response, I am aware that there are some bar only OSCs available but I have not found any I would like to use.  I was looking to keep my OSC looking like the default OSC but removing unwanted items.

@   Anyone
I am still interested in this if anyone has any ideas.
Are there even any tutorials on how this is done?
As mentioned I am just looking how to edit basic default items.

Thank You

**Comment by ahaoboy** [comment]
If you want to implement an OSC-like UI, you may be able to use mpv-easy, but you will need some front-end development experience.
For example, you can comment out unnecessary buttons. Compared to osc, mpv-easy is more modular.
https://github.com/mpv-easy/mpv-easy/blob/main/mpv-easy/src/ui/osc/control.tsx

**Comment by advice2020** [comment]
@ahaoboy 
Thank you for your response.
Unfortunately I do not.
I am just looking to slightly modify already existing things in the default osc.lua file


I might just be asking questions that are too complex / not possible
I thought because MPV has so many customization options and the fact that I imagine it has a huge user base that my questions would get tons of responses.
Does anyone know if the Discussions section here is considered to be the MPV forum or is there an official forum?
Not sure where else to ask these OSC customization questions, only looking to slightly modifying existing OSC, nothing fancy like other designers have achieved with modifying the OSC.

**Comment by ahaoboy** [comment]
To be honest, lua is not suitable for writing UI, and there are not many users of mpv, because it requires too many custom settings. You can try asking questions on [r](https://www.reddit.com/r/mpv/), where there are slightly more active users.

**Comment by ahaoboy** [comment]
Do you have a definite design for your custom UI? If you just want to cat some the osc UI element, I can put this case as a demo in the mpv-easy project

**Comment by advice2020** [comment]
Sorry I did not realize there was a response, thank you very much for this offer.
If possible was trying to stick to osc.lua especially because it is only a matter of moving / removing / modifying stuff that is already included in the default osc.lua file.
Because nothing is really being added, really just removing items which I have already been able to do and just needed help with background height and modifying the existing seekbar I was thinking there would be more people with the knowledge of how to achieve this, but as you mentioned maybe MPV does not have as many users as I had thought, at least not ones with the knowledge to achieve this either.
Anyways, thank you again for your offer

**Comment by advice2020** [comment]
`ADJUSTING BACKGROUND HEIGHT`
As mentioned I would like to be able to adjust the height of this black background rectangle.
I was able to figure out a little more about this but then got stuck again.
I narrowed the background down to the following code
```
local function bar_layout(direction)
    local osc_geo = {
        x = -2,
        y = nil,
        an = (direction < 0) and 7 or 1,
        w = nil,
        h = 56,
    }
```
The "h=56," appears to be the height of the black background rectangle. However if you change this to a lower value, it adjusts the height from the top down, so using a lower value then 56 ends up shortening the background from the bottom so you can see behind it, so I now need to move this background back down to the bottom window.
Does anyone know the code to edit that can be modified to make the black background move down so its edge comes in contact with the bottom of the MPV window?
Thank you for any help provided.

**Reply by JorgeR81** [reply]
@advice2020, to have a lower value instead of `h=56`, you also need to decrease the value in `line 2`, by the same amount.

For instance, to set `h=36`, you need change:

 `local line2 = osc_geo.y - direction * (36 + padY) ` 

to:
 `local line2 = osc_geo.y - direction * (16 + padY) `

---

I did this a long time ago in the mpv `0.32.0` OSC, but in still works the same way in `0.40.0`

![slimbar](https://github.com/user-attachments/assets/061b9d3f-93e1-4b81-82c0-1af17c477290)

**Reply by JorgeR81** [reply]
Hey, you figured out another solution ( https://github.com/mpv-player/mpv/discussions/14871 ). Nice !
But I think it's incomplete.
